Sentinel 2 - Level 1C eo3 prepare scripts

ohehir opened the pull request:
Merged sinergise and esa level 1C eo3 prepare scripts into a single script.

ohehir committed:
modified:   ../../../../eodatasets3/model.py
	new file:   ../../../../eodatasets3/prepare/sentinel_esa_L1_prepare.py
	new file:   S2B_MSIL1C_20201011T000249_N0209_R030_T55HFA_20201011T011446.zip
	new file:   ../../prepare/test_prepare_esa_sentinel_l1.py

ohehir committed:
Prelim esa level 1c support

jeremyh committed:
Fix zip path handling for s2

jeremyh committed:
pre-commit should check pep8 naming

ohehir committed:
new file:   sentinel_L1C_prepare.py
	modified:   ../../tests/integration/prepare/test_prepare_esa_sentinel_l1.py
	modified:   ../../tests/integration/prepare/test_prepare_sinergise_sentinel_l1.py
	deleted:    .landsat_l1_prepare.py.swp
	deleted:    sentinel_esa_L1_prepare.py
	deleted:    sentinel_sinergise_L1_prepare.py

ohehir committed:
modified:   test_prepare_sinergise_sentinel_l1.py
	deleted:    ../../../eodatasets3/prepare/.landsat_l1_prepare.py.swp
	deleted:    ../../../eodatasets3/prepare/sentinel_esa_L1_prepare.py
	deleted:    ../../../eodatasets3/prepare/sentinel_sinergise_L1_prepare.py

ohehir committed:
modified:   test_prepare_sinergise_sentinel_l1.py
	deleted:    ../../../eodatasets3/prepare/.landsat_l1_prepare.py.swp
	deleted:    ../../../eodatasets3/prepare/sentinel_esa_L1_prepare.py
	deleted:    ../../../eodatasets3/prepare/sentinel_sinergise_L1_prepare.py

ohehir committed:
set to ignore PVI layer

ohehir committed:
modified:   sentinel_L1C_prepare.py

ohehir committed:
deleted:    sentinel_esa_L1_prepare.py
	deleted:    sentinel_sinergise_L1_prepare.py

ohehir committed:
modified:   prepare.py

ohehir committed:
additional sentinel properties added to the dataset assembler's INHERITABLE_PROPERTIES dict

jeremyh commented on eodatasets3/prepare/sentinel_L1C_prepare.py:
```diff
@@ -0,0 +1,393 @@
+"""
```
Keep py file names lowercase, for consistency. 

(I'm surprised the pep8 linter doesn't complain about this)

jeremyh commented on eodatasets3/prepare/sentinel_L1C_prepare.py:
```diff
@@ -0,0 +1,393 @@
+    dataset_document: Path,
+) -> Tuple[uuid.UUID, Path]:
+    # Process esa dataset
+    if "zip" in str(dataset):
```
I'm a bit iffy about using 'zip' as the way to distinguish the type of dataset (ESA vs Synergise). From experience, people often try to run these things with unzipped folders or otherwise, and it's going to quietly try to parse all the metadata differently depending on whether someone has done that.

Is there a file inside that we could use to detect? Or a more specific part of the filename?

jeremyh commented on eodatasets3/prepare/sentinel_L1C_prepare.py:
```diff
@@ -0,0 +1,393 @@
+        }
+
+
+def process_metadata_xml(metadata_xml_path):
```
All the type hints from the old prepare script seem to be removed -- across all the functions in this file.

Eg, the old one: https://github.com/GeoscienceAustralia/eo-datasets/blob/eodatasets3/eodatasets3/prepare/sentinel_sinergise_L1_prepare.py#L56

(they're especially useful here because this file has a mix of `str` and `Path` variables that are all called "path", and this can be confusing to people. .... ideally, [make them all Paths](https://github.com/GeoscienceAustralia/eo-datasets/pull/130#discussion_r595605063)!)

jeremyh commented on eodatasets3/prepare/sentinel_L1C_prepare.py:
```diff
@@ -0,0 +1,393 @@
+    "--dataset",
+    type=PathPath(),
+    required=True,
+    help="Path to ESA zipped dataset or Sinergise dataset directory",
```
Nice, specific docs :)

jeremyh commented on eodatasets3/scripts/prepare.py:
```diff
@@ -13,7 +13,7 @@ def run():
 
 run.add_command(landsat_l1, name="landsat-l1")
-run.add_command(sin_sentinel_l1, name="sin-sentinel-l1")
+run.add_command(sentinel_l1c, name="sentinel-l1C")
```
```suggestion
run.add_command(sentinel_l1c, name="sentinel-l1")
```

Keep lowercase, like the others. And we probably don't need the `C`. 

... Landsat is just `landsat-l1`, so we my as well match it.

jeremyh reviewed (changes requested):
Thanks Aaron.

- We probably need to review the sentinel stac properties, but I might do this in a separate PR. (... some of these look more like ODC properties, in the `sentinel:` namespace [?] And others like 'orbit' probably have general standards already? We should avoid making up our own in anothers' namespace)
- There's a stray swap file included in a commit: https://github.com/GeoscienceAustralia/eo-datasets/blob/eabdc479958443ad7ed17f339ba9ccb75af62f09/eodatasets3/prepare/.landsat_l1_prepare.py.swp

(you can add `*,swp` to you `.gitignore` file to avoid accidentally committing these in the future, if your editor uses such names)

ohehir committed:
sentinel stac properties added

ohehir committed:
modified:   .gitignore

ohehir committed:
make sentinel_l1c_prepare.py lowercase

ohehir committed:
add type hints for functions

ohehir committed:
function names made lowercase ie. L1C changed to l1c

ohehir committed:
modified:   sentinel_l1c_prepare.py
	deleted:    sentinel_L1C_prepare.py
modified:   ../scripts/prepare.py

ohehir committed:
deleted:    sentinel_L1C_prepare.py
	modified:   ../scripts/prepare.py

jeremyh committed:
Remove intermediate property conversion

It's an unnecessary extra property name to
keep consistent in multiple places.

We're about to change properties so this
should make it easier.

jeremyh committed:
Update esa s2 test for stable grid names

After parallel PR #137 was merged.

jeremyh committed:
Use 'sat' extension, minimise property invention

jeremyh committed:
Clean-up of Sentinel L1 properties

- Use 'sat' Stac extension, not custom fields.
- Use the existing DEA sentinel props: tile_id, datatake time
- Minimise property invention: stick to ones we can see on other collection
- Fix incorrect datastrip_id value

The software version properties seemed too misleading to me: that's
the version of the QA software that was run, not the processing itself.

(if this is useful, we can re-add and name it something QA-related)

jeremyh committed:
Include accessory files in S2 L1

jeremyh committed:
Slightly more specific S2 input type test

With exception to give user better error message.

jeremyh committed:
Slightly simplify band path handling, s2

jeremyh committed:
Do not inherit all sentinel2 properties

jeremyh committed:
Check for more xml extraction errors

Raise exceptions when children don't match expected. Otherwise
we face silent errors.

jeremyh committed:
Auto-inherit useful Sentinel properties

jeremyh committed:
Include sat-extension in inherited properties

jeremyh commented:
- Did a review of our S2 properties here instead of as a new PR. Made them a little more consistent with our other products, and external Sentinel 2 stac products. We don't want to invent our own properties if we can help it.
- Updated the inheritable properties so that wagl product will get them by default too.

jeremyh commented:
There'll probably be more review before we finalise our properties, but this is good enough for develop.
